content/docs/connect/connection-pooling.md

```diff
@@ -49,9 +49,58 @@ The **Connection Details** widget on the Neon **Dashboard** provides **Pooled co
 
 ![Connection Details pooled connection string](/docs/connect/connection_details_pooled.png)
 
+## Optimize queries with PgBouncer and prepared statements
+
+Protocol-level prepared statements are supported with Neon and PgBouncer as of the [PgBouncer 1.21.0 release](https://github.com/pgbouncer/pgbouncer/releases/tag/pgbouncer_1_21_0). Using prepared statements can help boost query performance while providing an added layer of protection against potential SQL injection attacks.
+
+### Understanding prepared statements
+
+A prepared statement in Postgres allows for the optimization of an SQL query by defining its structure once and executing it multiple times with varied parameters. Here's an SQL-level example to illustrate, but please note that direct SQL-level `PREPARE` and `EXECUTE` are not supported with PgBouncer (see [below](#use-prepared-statements-with-pgbouncer)).
+
+```sql
+PREPARE fetch_plan (TEXT) AS
+SELECT * FROM users WHERE username = $1;
+
+EXECUTE fetch_plan('alice');
+```
+
+`fetch_plan` here is the prepared statement's name, and `$1` acts as a parameter placeholder.
+
+The benefits of using prepared statements include:
```

Comment:
Is there any caching supported? If you can save the prepared statement to cache, that can also speed things up.

Reply:
My understanding is that the caching of prepared statement query plans would be handled by the PostgreSQL planner and executor. There's no additional user action here.